tests/nat/nat_helpers.py

```diff
@@ -0,0 +1,848 @@
+            keys = output_cli["stdout"].split()[:5]
+            for num in range(0, num_entries):
+                entry_values = entries[(num * 5):(num * 5) + 5]
+                key = entry_values[1] if entry_values[1] != "---" else entry_values[2]
+                output[key] = {keys[i]: entry_values[i] for i in range(0, len(keys))}
+        return output
+    elif clear:
+        output_cli = exec_command(duthost, ["sudo sonic-clear nat statistics"])
+        if output_cli["rc"]:
+            raise Exception('Return code is {} not 0'.format(output_cli["rc"]))
+        return output_cli["stdout"].lstrip()
+    return None
+
+
+def dut_nat_iptables_status(duthost):
+    """
+    NAT CLI helper enable/disable DUT's interface
+    :param duthost: DUT host object
+    :return : dict with nat PREROUTING/POSTROUTING iptables entries
+    """
+    nat_table_status = {}
+    output_cli = exec_command(duthost, ["sudo iptables -nL -t nat"])
+    if output_cli["rc"]:
+        raise Exception('Return code is {} not 0'.format(output_cli["rc"]))
+    entries = output_cli["stdout"].split("\n")
+    index_prerouting = [i for i in range(0, len(entries)) if "PREROUTING" in entries[i]][0] + 2
+    index_input = [i for i in range(0, len(entries)) if "INPUT" in entries[i]][0]
+    index_postrouting = [i for i in range(0, len(entries)) if 'POSTROUTING' in entries[i]][0] + 2
+    if any(['DOCKER' in entry for entry in entries]):
+        index_docker = [i for i in range(0, len(entries)) if 'DOCKER' in entries[i]][0]
+        postrouting = [el for el in entries[index_postrouting:index_docker] if len(el) > 1]
+    else:
+        postrouting = [el for el in entries[index_postrouting:] if len(el) > 1]
+    prerouting = [el for el in entries[index_prerouting:index_input] if len(el) > 0]
+    nat_table_status["prerouting"] = [" ".join([s.strip() for s in el.split() if len(el) > 0])
+                                      for el in prerouting]
+    nat_table_status["postrouting"] = [" ".join([s.strip() for s in el.split() if len(el) > 0])
+                                       for el in postrouting]
+    return nat_table_status
+
+
+def dut_interface_status(duthost, interface_name):
+    """
+    NAT CLI helper enable/disable DUT's interface
+    :param duthost: DUT host object
+    :param interface_name: string interface to configure
+    :return : string formatted CLI output with interface current operstatus
+    """
+    interfaces_status = {}
+    output_cli = exec_command(duthost, ["show interfaces status | grep {}".format(interface_name)])
+    if output_cli["rc"]:
+        raise Exception('Return code is {} not 0'.format(output_cli["rc"]))
+    entries = output_cli["stdout"].split()
+    num_entries = len(entries[::10])
+    for num in range(0, num_entries):
+        entry_values = entries[(num * 10):(num * 10) + 10]
+        interfaces_status[entry_values[0]] = entry_values
+    return interfaces_status[interface_name][6]
+
+
```

Comment:
use the show_interface ansible module[[https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/show_interface.py](https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/show_interface.py)]

Reply:
Thanks, will be updated with next commit